subsys/bluetooth/mesh/access.c

```diff
@@ -486,24 +489,67 @@ struct bt_mesh_elem *bt_mesh_elem_find(uint16_t addr)
 {
 	uint16_t index;
 
+	if (!BT_MESH_ADDR_IS_UNICAST(addr)) {
+		return NULL;
+	}
+
+	index = addr - dev_comp->elem[0].addr;
+	if (index >= dev_comp->elem_count) {
+		return NULL;
+	}
+
+	return &dev_comp->elem[index];
+}
+
+bool bt_mesh_has_addr(uint16_t addr, bool trigger_msg_cb)
+{
+	uint16_t index;
+
 	if (BT_MESH_ADDR_IS_UNICAST(addr)) {
-		index = (addr - dev_comp->elem[0].addr);
-		if (index < dev_comp->elem_count) {
-			return &dev_comp->elem[index];
-		} else {
-			return NULL;
-		}
+		return bt_mesh_elem_find(addr) != NULL;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_MESH_ACCESS_LAYER_MSG) && trigger_msg_cb) {
```

Comment:
```suggestion
	if (IS_ENABLED(CONFIG_BT_MESH_ACCESS_LAYER_MSG) && trigger_msg_cb && msg_cb) {
```

Reply:
I can add this, and it makes sense to do so. I will have to remove the compiler check at the definition of the msg_cb function pointer though. But that's no problem.

Reply:
If the access to the variable is always protected by an IS_ENABLED call, the compiler will detect that it's logically impossible to access it, and remove it from the final image anyway, so no harm in this 🙂